File: с#/NeuronNetworks/Perseptron/Neurons/Neuron.cs
using System;
using NeuronNetworks.Perseptron.Interfaces;

namespace NeuronNetworks.Perseptron.Neurons
{
    public class Neuron: INeuron
    {
        static readonly Random random = new Random();

        public double[] Weights { get; }

        public ILayer CurrentLayer { get; set; }

        public int Index { get; }

        public Neuron(int weightLength, int index, ILayer currentLayer)
        {
            Weights = new double[weightLength];

            for (int i = 0; i < weightLength; i++)
            {
                Weights[i] = (double)1 / random.Next(2, 10);
            }

            CurrentLayer = currentLayer;
            Index = index;
        }

        public double Handle(params double[] inputs)
        {
            double power = 0;

            for(int i = 0; i < inputs.Length; i++)
            {
                power += inputs[i] * Weights[i];
            }

            return power.ThroughtSigmoid();
        }
    }
}

File: JavaScript (TypeScript)/src/neuron-network/neuron.ts
import { ILayer, INeuron } from './abstract';
import { Utils } from './util';

export class Neuron implements INeuron {
    public readonly weights: number[];

    public currentLayer: ILayer;

    public readonly index: number;

    private initializeWeights(weightLength: number): number[] {
        const weights = [];

        for (let i = 0; i < weightLength; i++) {
            weights[i] = 1 / Utils.randomInteger(2, 50);
        }

        return weights;
    }

    /**
     * Neuron
     */
    constructor(weightLength: number, index: number, currentLayer: ILayer) {
        this.weights = this.initializeWeights(weightLength);
        this.index = index;
        currentLayer = currentLayer;
    }

    handle(inputs: number[]): number {
        let power = 0;

        for (let i = 0; i < this.weights.length; i++) {
            power += this.weights[i] * inputs[i];
        }

        return power;
    }
}
